ledger/internal/eval_test.go

```diff
@@ -528,34 +528,34 @@ func (ledger *evalTestLedger) LookupWithoutRewards(rnd basics.Round, addr basics
 	return ledgercore.ToAccountData(ad), rnd, nil
 }
 
-func (ledger *evalTestLedger) LookupApplication(rnd basics.Round, addr basics.Address, aidx basics.AppIndex) (ledgercore.AppResource, error) {
+func (ledger *evalTestLedger) LookupApplication(addr basics.Address, aidx basics.AppIndex) (ledgercore.AppResource, basics.Round, error) {
 	res := ledgercore.AppResource{}
-	ad, ok := ledger.roundBalances[rnd][addr]
+	ad, ok := ledger.roundBalances[ledger.Latest()][addr]
 	if !ok {
-		return res, fmt.Errorf("no such account %s", addr.String())
+		return res, 0, fmt.Errorf("no such account %s", addr.String())
 	}
 	if params, ok := ad.AppParams[aidx]; ok {
 		res.AppParams = &params
 	}
 	if ls, ok := ad.AppLocalStates[aidx]; ok {
 		res.AppLocalState = &ls
 	}
-	return res, nil
+	return res, ledger.Latest(), nil
 }
 
-func (ledger *evalTestLedger) LookupAsset(rnd basics.Round, addr basics.Address, aidx basics.AssetIndex) (ledgercore.AssetResource, error) {
+func (ledger *evalTestLedger) LookupAsset(addr basics.Address, aidx basics.AssetIndex) (ledgercore.AssetResource, basics.Round, error) {
 	res := ledgercore.AssetResource{}
-	ad, ok := ledger.roundBalances[rnd][addr]
+	ad, ok := ledger.roundBalances[ledger.Latest()][addr]
 	if !ok {
-		return res, fmt.Errorf("no such account %s", addr.String())
+		return res, 0, fmt.Errorf("no such account %s", addr.String())
 	}
 	if params, ok := ad.AssetParams[aidx]; ok {
 		res.AssetParams = &params
 	}
 	if h, ok := ad.Assets[aidx]; ok {
 		res.AssetHolding = &h
 	}
-	return res, nil
+	return res, ledger.Latest(), nil
```

Comment:
calling ledger.Latest() twice seems dangerous — could we add `rnd := ledger.Latest()` at the top and return it